src/helix/client/client_ext.rs

```diff
@@ -1070,27 +1202,29 @@ impl<'client, C: crate::HttpClient + Sync + 'client> HelixClient<'client, C> {
     /// use twitch_api::helix;
     ///
     /// let rewards: Vec<helix::points::CustomReward> = client
-    ///     .get_custom_rewards("1234", true, &["8969ec47-55b6-4559-a8fe-3f1fc4e6fe58".into()], &token)
+    ///     .get_custom_rewards("1234", true, &["8969ec47-55b6-4559-a8fe-3f1fc4e6fe58"][..].into(), &token)
     ///     .await?;
-    ///
     /// # Ok(()) }
     /// ```
+    // XXX: This function is useless as a stream, since you can never have more than 50 rewards on a channel
     pub async fn get_custom_rewards<'b, T>(
         &'client self,
         broadcaster_id: impl types::IntoCow<'b, types::UserIdRef> + Send + 'b,
         only_managable_rewards: bool,
-        // FIXME: This should be `impl AsRef<[&'b T]> + 'b`
-        ids: &'b [&'b types::RewardIdRef],
-        token: &T,
+        ids: &'b types::Collection<'b, types::RewardId>,
+        token: &'client T,
     ) -> Result<Vec<helix::points::CustomReward>, ClientError<C>>
     where
         T: TwitchToken + Send + Sync + ?Sized,
     {
+        if ids.len() > 50 {
+            return Err(ClientRequestError::Custom("too many IDs, max 50".into()));
+        }
```

Comment:
I wonder if the library should enforce this (same goes for other checks). If Twitch changes the limit, users would need to update the library.

Reply:
I removed this check, there should not be any other left now